internal/dbx/measurement_test.go

```diff
@@ -0,0 +1,338 @@
+				CreatedAt: testutil.NewPgxTimestamp(time.Date(2025, time.March, 1, 5, 0, 0, 0, utc)),
+				// Next month in UTC and ET
+			},
+		},
+		Resources: []db.Resource{
+			{
+				Meter:         meterName,
+				NaturalID:     resourceID,
+				KindNaturalID: kindID,
+				CFOrgID:       orgID,
+			},
+		},
+		Measurements: []db.Measurement{
+			{
+				Meter:             meterName,
+				ResourceNaturalID: resourceID,
+				Value:             7,
+				ReadingID:         readingID1,
+			},
+			{
+				Meter:             meterName,
+				ResourceNaturalID: resourceID,
+				Value:             7,
+				ReadingID:         readingID2,
+			},
+			{
+				Meter:             meterName,
+				ResourceNaturalID: resourceID,
+				Value:             7,
+				ReadingID:         readingID3,
+			},
+			{
+				Meter:             meterName,
+				ResourceNaturalID: resourceID,
+				Value:             7,
+				ReadingID:         readingID4,
+			},
+			{
+				Meter:             meterName,
+				ResourceNaturalID: resourceID,
+				Value:             7,
+				ReadingID:         readingID5,
+			},
+			{
+				Meter:             meterName,
+				ResourceNaturalID: resourceID,
+				Value:             7,
+				ReadingID:         readingID6,
+			},
+		},
+	}
+
+	for _, i := range td.Orgs {
+		_, err := q.CreateCFOrg(t.Context(), i.ID)
+		if err != nil {
+			t.Fatal("creating CF org failed", err)
+		}
+	}
+	for _, i := range td.Meters {
+		_, err := q.CreateMeter(t.Context(), i.Name)
+		if err != nil {
+			t.Fatal("creating meter failed", err)
+		}
+	}
+	for _, i := range td.Kinds {
+		_, err := q.CreateResourceKind(t.Context(), db.CreateResourceKindParams{
+			Meter:     i.Meter,
+			NaturalID: i.NaturalID,
+		})
+		if err != nil {
+			t.Fatal("creating resource kind failed", err)
+		}
+	}
+	for _, i := range td.Prices {
+		_, err := q.CreatePriceWithID(t.Context(), db.CreatePriceWithIDParams(i))
+		if err != nil {
+			t.Fatal("creating price failed", err)
+		}
+	}
+	for _, i := range td.Readings {
+		_, err := q.CreateReadingWithID(t.Context(), db.CreateReadingWithIDParams{
+			ID:        i.ID,
+			CreatedAt: i.CreatedAt,
+			Periodic:  i.Periodic,
+		})
+		if err != nil {
+			t.Fatal("creating reading failed", err)
+		}
+	}
+	for _, i := range td.Resources {
+		err = q.CreateResources(t.Context(), db.CreateResourcesParams(i))
+		if err != nil {
+			t.Fatal("creating resource failed", err)
+		}
+	}
+	for _, i := range td.Measurements {
+		q.CreateMeasurement(t.Context(), db.CreateMeasurementParams{
+			ReadingID:         i.ReadingID,
+			Meter:             i.Meter,
+			ResourceNaturalID: i.ResourceNaturalID,
+			Value:             i.Value,
+		})
+	}
+
+	// Act
+	updated, err := q.UpdateMeasurementMicrocredits(t.Context(), asOf)
+
+	// Assert
+	if err != nil {
+		t.Fatal("error occured while calling function under test", err)
+	}
+	if updated.Int64 != 3 {
+		t.Fatalf("expected %v rows updated, got %v", 2, updated.Int64)
+	}
+	ms, err := q.ListMeasurements(t.Context())
```

Comment:
suggestion: missed handling this error I think?
```suggestion
	if ms, err := q.ListMeasurements(t.Context(); err != nil {
		t.Fatalf("error occurred listing measurements: %v", err)
	}
```